cfme/tests/automate/test_file_import.py

```diff
@@ -201,3 +203,76 @@ def test_automate_import_attributes_updated(setup_automate_model, import_datasto
     assert view.description.read() == "test_name_desc_updated"
     view = navigate_to(klass, "Edit")
     assert view.description.read() == "test_class_desc"
+
+
+@pytest.fixture(scope='module')
+def local_domain(appliance):
+    """This fixture used to create automate domain - Datastore/Domain"""
+    # Domain name should be static to match with name of domain imported using rake command
+    domain = appliance.collections.domains.create(
+        name="bz_1753860",
+        description=fauxfactory.gen_alpha(),
+        enabled=True)
+    yield domain
+    domain.delete_if_exists()
```

Comment:
I think it would be better if you use REST to delete the domain.
```
if domain.rest_api_entity.exists:
    domain.rest_api_entity.action.delete()
```
_The reason being_, if the test case fails/errors out/stops after disabling the domain, it will not be able to delete the domain via UI, since domain name in the tree would change to `domain_name (Disabled)`, and its tree path would also change to `["Datastore", "domain_name (Disabled)"].

To delete a domain, it first checks the path in the bootstrap tree. Since `.enabled` property of `local_domain` does not get updated when you enable or disable it and since the original domain.enabled value is `True`, it tries to search for ["Datastore", local_domain.name] in the bootstrap tree path, but does not find it and hence is not able to delete the domain.

With REST this problem does not arise, so it doesn't matter whether a domain is enabled or disabled, it will delete it if it exists as long as it is not locked.

Reply:
Thank you for detailed solution @valaparthvi 
It is very helpful. But it worked with 5.11 and givin some error in 5.10 as below:
```
ipdb> domain.rest_api_entity.action.all                                                              
['refresh_from_source']
```
It does not have delete action in 5.10 via REST

Reply:
Oh yes, so maybe in that case, instead of using REST, you could make sure that the domain is enabled before deleting it, or maybe modify the delete() method to delete a domain no matter it's status.

Reply:
Another solution would be something like this - 
```suggestion
@pytest.fixture(scope='module')
def local_domain(appliance):
    """This fixture used to create automate domain - Datastore/Domain"""
    # Domain name should be static to match with name of domain imported using rake command
    domain = appliance.collections.domains.create(
        name="bz_1753860", description=fauxfactory.gen_alpha(), enabled=True
    )
    yield domain
    
    domain.enabled = domain.rest_api_entity.enabled
    
    domain.delete()
```

The reason being, `treepath` of a domain is decided on the basis of it's `enabled` status. We can change `domain.enabled` to whatever it's current value is, that way the domain will be deleted for sure.